### PR TITLE
improve cygwin path expansion

### PR DESCRIPTION
**Pull request by ObiWahn**

### CODE REVIEW

File: containers/documentation.docker/build_documentation [hunk 1]
```diff
@@ -20,22 +20,28 @@
...
+                out+=("$(cygpath -w "$path")")
+                continue
+            fi
+        fi
+        out+=("$path")
+    done
+    echo "${out[@]}"
 }
 
+echo $(expand_cygpath "/cygwin/c/lala" "moepse" "/cygwin/c/yg/win")
```
Comment (Simran-B) [inline containers/documentation.docker/build_documentation]: I guess this was for testing only?

File: containers/documentation.docker/build_documentation [hunk 2]
```diff
@@ -20,22 +20,28 @@
 ## get script dir
 script_dir="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
 
-# determine if this is run in Cygwin under Windows
-is_cygwin=false
-if [[ "$(uname -o || true)" == "Cygwin" ]]; then
-    is_cygwin=true
-fi
-
 # Convert to Windows path in Cygwin, i.e. '/cygwin/c/...' to 'C:\...'
```
Comment (Simran-B) [inline containers/documentation.docker/build_documentation]: My example is wrong, it's `/cygdrive/c/...`

File: containers/documentation.docker/build_documentation [hunk 3]
```diff
@@ -20,22 +20,28 @@
...
-    local any_path="$1"
-    if [[ -n $any_path ]] && $is_cygwin; then
-        echo "$(cygpath -w $any_path)"
-    else
-        echo "${any_path}"
-    fi
+    local path
+    local out=()
+    for path in "$@"; do
+        local matchre='/cygwin/'
```
Comment (Simran-B) [inline containers/documentation.docker/build_documentation]: Should be `'^/cygdrive/'` I guess

File: containers/documentation.docker/build_documentation [hunk 4]
```diff
@@ -20,22 +20,28 @@
...
+        if type cygpath 2>/dev/null; then
```
Comment (Simran-B) [inline containers/documentation.docker/build_documentation]: Does this catch empty args?

```
$ cygpath -w ""
cygpath: can't convert empty path
```

Even with no cache path it should never be empty ("fake cache" is passed in that case), but there is a check in the cache code that tests for empty string as well.

File: containers/documentation.docker/build_documentation (hunk 1 above)
Comment (ObiWahn) [reply containers/documentation.docker/build_documentation]: yes it was meant to show you how it works - i think there is also an exit, that should not be in the final version

File: containers/documentation.docker/build_documentation (hunk 3 above)
Comment (ObiWahn) [reply containers/documentation.docker/build_documentation]: Yes please try the expansion on cygwin host!

File: containers/documentation.docker/build_documentation (hunk 4 above)
Comment (ObiWahn) [reply containers/documentation.docker/build_documentation]: It is meant to take arbitrary arguments including an empty string, that way all arguments `fs paths` or not can be handled with one call.

Comment (ObiWahn) [reply containers/documentation.docker/build_documentation]: The RE should make sure that there is no empty string passed to the cypath command.

**Pull request merged**